AK/Stream.h

```diff
@@ -104,6 +105,22 @@ InputStream& operator<<(InputStream& stream, BigEndian<T> value)
     return stream;
 }
 
+template<typename T>
+InputStream& operator>>(InputStream& stream, Optional<T>& value)
+{
+    T temporary;
+    stream >> temporary;
+    value = temporary;
+    return stream;
+}
+template<typename T>
+OutputStream& operator<<(OutputStream& stream, const Optional<T>& value)
+{
+    if (value.has_value())
+        return stream << value.value();
+    return stream;
+}
```

Comment:
This feels like a very weird API. It may or may not write to the stream, depending of whether the optional is empty or not.

Reply:
Initially, I had the same thought but then I dismissed it, but I have to agree it's pretty weird and I can't think of a use case. I've rebased and removed it.